Modules/IO/MeshBase/include/itkMeshIOTestHelper.h

```diff
@@ -398,7 +398,7 @@ AllocateBuffer(void ** data, itk::IOComponentEnum componentType, itk::SizeValueT
       buffer = new unsigned int[bufferSize];
       break;
     case itk::IOComponentEnum::INT:
-      buffer = new unsigned int[bufferSize];
+      buffer = new int[bufferSize];
```

Comment:
Looks like a proper fix to me.  In this particular case (`IOComponentEnum::INT`), `int` must have been intended, instead of `unsigned int`. However, I don't expect any observable behavior change. In C++, a pointer to `unsigned int` may be converted to a void pointer, and then casted to and "interpreted" as a pointer to signed `int`, without runtime errors.

Reply:
I look from time to time at _Valgrind_ builds. IMHO, the whole approach, that `AllocateBuffer`  and `void**`, should be reviewed. Exactly after #3403 the amount of _Valgrind_ defects went from 1 to 33, they are all from mesh tests and about the allocation helper.

[30.05.2022](https://open.cdash.org/viewDynamicAnalysis.php?buildid=7946542)

[31.05.2022](https://open.cdash.org/viewDynamicAnalysis.php?buildid=7946542)

The errors are like this:
```
FMM ==16981== Mismatched free() / delete / delete []
==16981==    at 0x483CFBF: operator delete(void*) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==16981==    by 0x1FF10D: itkOBJMeshIOTest(int, char**) (itkOBJMeshIOTest.cxx:183)
==16981==    by 0x12ED77: main (ITKIOMeshOBJTestDriver.cxx:142)
==16981==  Address 0x89e4ab0 is 0 bytes inside a block of size 400,000 alloc'd
==16981==    at 0x483C583: operator new[](unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==16981==    by 0x1FB27A: AllocateBuffer(void**, itk::CommonEnums::IOComponent, unsigned long) (itkMeshIOTestHelper.h:416)
==16981==    by 0x1FDB56: itkOBJMeshIOTest(int, char**) (itkOBJMeshIOTest.cxx:106)
==16981==    by 0x12ED77: main (ITKIOMeshOBJTestDriver.cxx:142)
```

Reply:
Fixed typo in links to builds.